metrics/web/test/auth/presentation/state/auth_notifier_test.dart

```diff
@@ -235,56 +248,110 @@ void main() {
       verifyNever(googleSignInUseCase());
     });
 
-    test(".authErrorMessage is populated when SignInUseCase throws", () {
+    test(".authErrorMessage is populated when SignInUseCase throws", () async {
       when(signInUseCase.call(any)).thenThrow(authException);
 
-      authNotifier.signInWithEmailAndPassword(email, password);
+      await authNotifier.signInWithEmailAndPassword(email, password);
 
       expect(authNotifier.authErrorMessage, isNotNull);
     });
 
-    test(".authErrorMessage is populated when GoogleSignInUseCase throws", () {
-      when(googleSignInUseCase.call()).thenThrow(authException);
+    test(
+      ".emailErrorMessage is populated when the email-related error occurred during the sign-in process",
+      () async {
+        when(signInUseCase.call(any)).thenThrow(emailAuthException);
```

Comment:
Should we use the `thenAnswer` method since this is an async use case? The same for all tests below.